server/neptune/workflows/internal/activities/terraform.go

```diff
@@ -122,10 +122,48 @@ func (t *terraformActivities) TerraformPlan(ctx context.Context, request Terrafo
 // Terraform Apply
 
 type TerraformApplyRequest struct {
+	Args      []terraform.Argument
+	Envs      map[string]string
+	JobID     string
+	TfVersion string
+	Path      string
+}
+
+type TerraformApplyResponse struct {
+	Output string
+}
+
+func (t *terraformActivities) TerraformApply(ctx context.Context, request TerraformApplyRequest) (TerraformApplyResponse, error) {
+	// Fail requests using a target flag, as Atlantis cannot support this functionality
+	if containsTargetFlag(request.Args) {
+		return TerraformApplyResponse{}, errors.New("request contains invalid -target flag")
+	}
+
+	tfVersion, err := t.resolveVersion(request.TfVersion)
+	if err != nil {
+		return TerraformApplyResponse{}, err
+	}
+
+	planFile := buildPlanFilePath(request.Path)
+	args := []terraform.Argument{DisableInputArg}
+	args = append(args, request.Args...)
+
+	cmd := terraform.NewSubCommand(terraform.Apply).WithInput(planFile).WithArgs(args...)
+	ch := t.TerraformClient.RunCommand(ctx, request.JobID, request.Path, cmd, request.Envs, tfVersion)
+	_, err = t.readCommandOutput(ch)
+	if err != nil {
+		return TerraformApplyResponse{}, errors.Wrap(err, "processing command output")
+	}
+	return TerraformApplyResponse{}, nil
 }
 
-func (t *terraformActivities) TerraformApply(ctx context.Context, request TerraformApplyRequest) error {
-	return nil
+func containsTargetFlag(args []terraform.Argument) bool {
+	for _, arg := range args {
+		if arg.Key == "target" {
+			return true
+		}
+	}
+	return false
 }
```

Comment:
do we need this really right now? I don't think request.Args is even connected to anything or will be ever since it is run through github checks

Reply:
I think request.Args comes from gateway root building, so I left it there in the instance a user adds that to a custom step configuration. We can remove it later if we think it's no longer relevant.

Reply:
the only users that change that are us? so its pretty irrelevant imo

Reply:
Also this type of validation should be something we catch as part of parsing if anything, not all the way down here.